qiskit_machine_learning/neural_networks/circuit_qnn.py

```diff
@@ -208,6 +213,10 @@ def _sample(self, input_data: Optional[np.ndarray], weights: Optional[np.ndarray
 
     def _probabilities(self, input_data: Optional[np.ndarray], weights: Optional[np.ndarray]
                        ) -> Union[np.ndarray, SparseArray]:
+
+        if not self._quantum_instance:
+            raise QiskitMachineLearningError(
+                'Evaluation of probabilities requires a quantum instance!')
```

Comment:
If don't miss anything such checks should be in `_sample()`, `_probabilities()`, and `_probability_gradients()`. I guess `self._sampler.convert()` will raise an exception if `q_i` is `None`.